jsettlers.graphics/src/main/java/jsettlers/graphics/map/draw/MapObjectDrawer.java

```diff
@@ -256,37 +253,28 @@ public void drawShip(int x, int y, IMovable ship) {
 					numberOfFreight = maxNumberOfPassengers;
 				}
 				for (int i = 0; i < numberOfFreight; i++) {
-					yPosition.add(PASSENGER_POSITION_TO_FRONT[i] * yShiftForward + PASSENGER_POSITION_TO_RIGHT[i] * yShiftRight);
-					drawSequence.add(i);
+					freightY.add(new FloatIntObject(PASSENGER_POSITION_TO_FRONT[i] * yShiftForward
+							+ PASSENGER_POSITION_TO_RIGHT[i] * yShiftRight, i));
 				}
 			} else {
 				numberOfFreight = ship.getNumberOfStacks();
 				if (numberOfFreight > maxNumberOfStacks) {
 					numberOfFreight = maxNumberOfStacks;
 				}
 				for (int i = 0; i < numberOfFreight; i++) {
-					yPosition.add(CARGO_POSITION_TO_FRONT[i] * yShiftForward + CARGO_POSITION_TO_RIGHT[i] * yShiftRight);
-					drawSequence.add(i);
+					freightY.add(new FloatIntObject(CARGO_POSITION_TO_FRONT[i] * yShiftForward
+							+ CARGO_POSITION_TO_RIGHT[i] * yShiftRight, i));
 				}
 			}
 			// sort freight by view y
-			for (int i = 0; i < numberOfFreight - 1; i++) {
-				for (int j = i + 1; j < numberOfFreight; j++) {
-					if (yPosition.get(i) < yPosition.get(j)) {
-						int number = drawSequence.get(i);
-						float yPos = yPosition.get(i);
-						drawSequence.set(i, drawSequence.get(j));
-						yPosition.set(i, yPosition.get(j));
-						drawSequence.set(j, number);
-						yPosition.set(j, yPos);
-					}
-				}
+			if (freightY.size() > 0) {
+				Collections.sort(freightY, freightY.get(0));
```

Comment:
Mind the difference between `Comparator`s and `Comparable`s.
* A `Comparator` takes two objects and compares them. A comparator has no internal state. It is a class implementing just one method. You could use a Lambda or a new class for this, but you should not abuse any normal object for this. Java allows you to create anonymous classes, so your second parameter could be `new Comparator() { ... }`
* `Comparable` is an interface implemented by objects that have a natural order. `compareTo` compares the current item to any other item. This is meant for numbers, strings, ... where the order is obvious.